sdk/core/azure-core/src/datetime.cpp

```diff
@@ -0,0 +1,792 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/core/datetime.hpp"
+
+#include <cstdio>
+#include <cstring>
+#include <limits>
+
+#ifdef _WIN32
+#define WIN32_LEAN_AND_MEAN
+#define NOMINMAX
+#include <windows.h>
+#else
+#include <sys/time.h>
+#endif
```

Comment:
we have WINDOWS and UNIX defined for the project.
This helps with MINGW , MSYS and CYGWIN

Also, if not Windows or Lin, the else will fail?  should you throw for other platform?

Reply:
It won't compile, if `sys/time.h` is not present. But it does on POSIX platforms, which includes MacOS: https://en.wikipedia.org/wiki/C_POSIX_library

Reply:
As of WINDOWS and UNIX, I think if we avoid relying on cmake, it is even better.